codec.go

```diff
@@ -79,10 +84,17 @@ func (p protoCodec) Marshal(v interface{}) ([]byte, error) {
 }
 
 func (p protoCodec) Unmarshal(data []byte, v interface{}) error {
+	protoMsg := v.(proto.Message)
+	protoMsg.Reset()
```

Comment:
Per `proto.Unmarshaler`: "The method should reset the receiver before decoding starts."

So this should not be necessary above the call to `Unmarshal`.  Also, `buffer.Unmarshal` says "Unlike proto.Unmarshal, this does not reset pb before starting to unmarshal."

Reply:
I looked at some of our Unmarshalers generated via gogoproto and they don't reset themselves in their `Unmarshal`. I think it would be dangerous to not reset here. See https://github.com/gogo/protobuf/issues/334 and https://github.com/golang/protobuf/issues/424.

Reply:
> See ... golang/protobuf#424.

That's...amazing...

OK, this LGTM then.  Thanks!